src/core/internal/string.d

```diff
@@ -76,7 +75,10 @@ Returns:
 */
 auto unsignedToTempString(uint radix = 10)(ulong value) @safe
 {
-    TempStringNoAlloc!() result = void;
+    // Need a buffer of 65 bytes for radix of 2 with room for
+    // signedToTempString to possibly add a negative sign.
+    enum bufferSize = radix >= 10 ? 20 : 65;
+    TempStringNoAlloc!bufferSize result = void;
```

Comment:
Hm... technically, we could exactly calculate the sizes. But this is a reasonable cutoff.

Reply:
Yeah a balancing consideration in my mind was having fewer distinct instantiations of `TempStringNoAlloc`, especially if (somehow) the return type of this function could end up being used to instantiate a cascade of other templates.

Reply:
Possibly another level for octal would be useful, as it would be 1/4 the size of the 65-byte thing, and it's a possible common use case. But it's theoretical at this point (I'm 99% sure we only ever do decimal and hexadecimal), if it turns out someone wants it they can add it.

Reply:
I recall the number of digits required to represent a number with `n` bits in base `k` is `1 + uint(n / log2(k))`. We have a `log2` thing in `allocator` that should work during compilation.

Reply:
Realistically, I think the only bases people will likely use are 2, 8, 10, and 16. If you allocate 23 bytes (I think), you cover base 8 or higher. If you truly wanted to optimize, 16 only requires 17 bytes. So I think the diminishing returns probably aren't worth the extra cost of having different struct instantiations.

But 23 or 20 instead of 65 is definitely a lot more efficient.